tensorflow/python/keras/utils/vis_utils.py

```diff
@@ -124,6 +147,26 @@ def model_to_dot(model,
     dot.set('dpi', dpi)
     dot.set_node_defaults(shape='record')
 
+  if layer_range is None:
+    layer_range = [0, len(model.layers)]
+  
+  # check point for layer range whether its string or integer 
+  # layer range must be same dtype (mixed dtype is not allowed)
+  if all(map(lambda x: str(x).isdigit(), layer_range)):
```

Comment:
I would suggest to move the error case here for checking the mixed types, and return early. So you could do

if len(set([type(x) for x in layer_range])) > 1:
  raise

Reply:
done